test/functional/page_objects/discover_page.ts

```diff
@@ -868,4 +868,51 @@ export class DiscoverPageObject extends FtrService {
     await this.fieldEditor.save();
     await this.header.waitUntilLoadingHasFinished();
   }
+
+  private async waitForDropToFinish() {
+    await this.retry.try(async () => {
+      const exists = await this.find.existsByCssSelector('.domDragDrop-isActiveGroup');
+      if (exists) {
+        throw new Error('UI still in drag/drop mode');
+      }
+    });
+    await this.header.waitUntilLoadingHasFinished();
+    await this.waitUntilSearchingHasFinished();
+  }
+
+  /**
+   * Drags field to add as a column
+   *
+   * @param fieldName
+   * */
+  public async dragFieldToTable(fieldName: string) {
+    await this.waitUntilSidebarHasLoaded();
+
+    const from = `dscFieldListPanelField-${fieldName}`;
+    await this.find.existsByCssSelector(from);
+    await this.browser.html5DragAndDrop(
+      this.testSubjects.getCssSelector(from),
+      this.testSubjects.getCssSelector('dscMainContent')
+    );
+    await this.waitForDropToFinish();
+  }
+
+  /**
+   * Drags field with keyboard actions to add as a column
```

Comment:
Glad to see this added to the test suite!

Reply:
@1Copenut Thank you very much for a such detailed feedback! I created a separate issue for a follow up work https://github.com/elastic/kibana/issues/155015